fix(vscode-web): remove exit if extension installation fails

matifali opened the pull request:
Resolves #317

code-asher reviewed (approved):
Makes sense!

code-asher commented on vscode-web/run.sh:
```diff
@@ -72,14 +72,12 @@ for extension in "$${EXTENSIONLIST[@]}"; do
 
 if [ "${AUTO_INSTALL_EXTENSIONS}" = true ]; then
   if ! command -v jq > /dev/null; then
     echo "jq is required to install extensions from a workspace file."
```
Oh wait we still need some kind of exit here.  Maybe put the block under an `else`?

matifali replied:
I guess it is ok to just give the message without exiting so that the script can continue.

code-asher replied:
Let me know if I misunderstood though :laughing: 

```
if [ "${AUTO_INSTALL_EXTENSIONS}" = true ]; then
  if ! command -v jq > /dev/null; then
    echo "jq is required to install extensions from a workspace file."
  else
   WORKSPACE_DIR="$HOME"
   if [ -n "${FOLDER}" ]; then
     WORKSPACE_DIR="${FOLDER}"
   fi

   if [ -f "$WORKSPACE_DIR/.vscode/extensions.json" ]; then
     printf "🧩 Installing extensions from %s/.vscode/extensions.json...\n" "$WORKSPACE_DIR"
     extensions=$(jq -r '.recommendations[]' "$WORKSPACE_DIR"/.vscode/extensions.json)
     for extension in $extensions; do
       $VSCODE_WEB "$EXTENSION_ARG" --install-extension "$extension" --force
     done
   fi
  fi
fi
```

code-asher reviewed (approved):
Awesome!

matifali commented:
OK. Looks like MS broke something when I run, `/tmp/vscode-web/bin/code-server --install-extension github.copilot --force` or `/tmp/vscode-web/bin/code-server --install-extension github.copilot` in the workspace using this module. I get;

```console
Installing extensions...
Extension 'github.copilot' is already installed.
file:///tmp/vscode-web/out/vs/server/node/server.main.js:33
        | KEY ('==' | '!=' | '<' | '<=' | '>' | '>=' | 'in' | 'not' 'in') value`,this.D())}}u(){const t=this.D();switch(t.type){case 17:case 18:return this.z(),t.lexeme;case 11:return this.z(),"true";case 12:return this.z(),"false";case 13:return this.z(),"in";default:return""}}w(t){return t.replaceAll(this.v,"")}x(){return this.f[this.g-1]}y(t){return this.C(t)?(this.z(),!0):!1}z(){return this.E()||this.g++,this.x()}A(t,s){if(this.C(t))return this.z();throw this.B(s,this.D())}B(t,s,i){const r=m(1692,null,t,wo.getLexeme(s)),n=s.offset,o=wo.getLexeme(s);return this.h.push({message:r,offset:n,lexeme:o,additionalInfo:i}),Sa.c}C(t){return this.D().type===t}D(){return this.f[this.g]}E(){return this.D().type===20}},xt=class{static false(){return qt.INSTANCE}static true(){return ns.INSTANCE}static has(e){return pn.create(e)}static equals(e,t){return yo.create(e,t)}static notEquals(e,t){return Eo.create(e,t)}static regex(e,t){return xo.create(e,t)}static in(e,t){return ih.create(e,t)}static notIn(e,t){return rh.create(e,t)}static not(e){return mn.create(e)}static and(...e){return bm.create(e,null,!0)}static or(...e){return nh.create(e,null,!0)}static greater(e,t){return Qa.create(e,t)}static greaterEquals(e,t){return Za.create(e,t)}static smaller(e,t){return Ya.create(e,t)}static smallerEquals(e,t){return Xa.create(e,t)}static{this.c=new v2({regexParsingWithErrorRecovery:!1})}static deserialize(e){return e==null?void 0:this.c.parse(e)}};function bo(e,t){return e.cmp(t)}var qt=class L6{static{this.INSTANCE=new L6}constructor(){this.type=0}cmp(t){return this.type-t.type}equals(t){return t.type===this.type}substituteConstants(){return this}evaluate(t){return!1}serialize(){return"false"}keys(){return[]}map(t){return this}negate(){return ns.INSTANCE}},ns=class O6{static{this.INSTANCE=new O6}constructor(){this.type=1}cmp(t){return this.type-t.type}equals(t){return t.type===this.type}substituteConstants(){return this}evaluate(t){return!0}serialize(){return"true"}keys(){return[]}map(t){return this}negate(){return qt.INSTANCE}},pn=class C6{static create(t,s=null){const i=Et.get(t);return typeof i=="boolean"?i?ns.INSTANCE:qt.INSTANCE:new C6(t,s)}constructor(t,s){this.key=t,this.c=s,this.type=2}cmp(t){return t.type!==this.type?this.type-t.type:ym(this.key,t.key)}equals(t){return t.type===this.type?this.key===t.key:!1}substituteConstants(){const t=Et.get(this.key);return typeof t=="boolean"?t?ns.INSTANCE:qt.INSTANCE:this}evaluate(t){return!!t.getValue(this.key)}serialize(){return this.key}keys(){return[this.key]}map(t){return t.mapDefined(this.key)}negate(){return this.c||(this.c=mn.create(this.key,this)),this.c}},yo=class M6{static create(t,s,i=null){if(typeof s=="boolean")return s?pn.create(t,i):mn.create(t,i);const r=Et.get(t);return typeof r=="boolean"?s===(r?"true":"false")?ns.INSTANCE:qt.INSTANCE:new M6(t,s,i)}constructor(t,s,i){this.c=t,this.d=s,this.f=i,this.type=4}cmp(t){return t.type!==this.type?this.type-t.type:Lr(this.c,this.d,t.c,t.d)}equals(t){return t.type===this.type?this.c===t.c&&this.d===t.d:!1}substituteConstants(){const t=Et.get(this.c);if(typeof t=="boolean"){const s=t?"true":"false";return this.d===s?ns.INSTANCE:qt.INSTANCE}return this}evaluate(t){return t.getValue(this.c)==this.d}serialize(){return`${this.c} == '${this.d}'`}keys(){return[this.c]}map(t){return t.mapEquals(this.c,this.d)}negate(){return this.f||(this.f=Eo.create(this.c,this.d,this)),this.f}},ih=class F6{static create(t,s){return new F6(t,s)}constructor(t,s){this.d=t,this.f=s,this.type=10,this.c=null}cmp(t){return t.type!==this.type?this.type-t.type:Lr(this.d,this.f,t.d,t.f)}equals(t){return t.type===this.type?this.d===t.d&&this.f===t.f:!1}substituteConstants(){return this}evaluate(t){const s=t.getValue(this.f),i=t.getValue(this.d);return Array.isArray(s)?s.includes(i):typeof i=="string"&&typeof s=="object"&&s!==null?c2.call(s,i):!1}serialize(){return`${this.d} in '${this.f}'`}keys(){return[this.d,this.f]}map(t){return t.mapIn(this.d,this.f)}negate(){return this.c||(this.c=rh.create(this.d,this.f)),this.c}},rh=class T6{static create(t,s){return new T6(t,s)}constructor(t,s){this.d=t,this.f=s,this.type=11,this.c=ih.create(t,s)}cmp(t){return t.type!==this.type?this.type-t.type:this.c.cmp(t.c)}equals(t){return t.type===this.type?this.c.equals(t.c):!1}substituteConstants(){return this}evaluate(t){return!this.c.evaluate(t)}serialize(){return`${this.d} not in '${this.f}'`}keys(){return this.c.keys()}map(t){return t.mapNotIn(this.d,this.f)}negate(){return this.c}},Eo=class U6{static create(t,s,i=null){if(typeof s=="boolean")return s?mn.create(t,i):pn.create(t,i);const r=Et.get(t);return typeof r=="boolean"?s===(r?"true":"false")?qt.INSTANCE:ns.INSTANCE:new U6(t,s,i)}constructor(t,s,i){this.c=t,this.d=s,this.f=i,this.type=5}cmp(t){return t.type!==this.type?this.type-t.type:Lr(this.c,this.d,t.c,t.d)}equals(t){return t.type===this.type?this.c===t.c&&this.d===t.d:!1}substituteConstants(){const t=Et.get(this.c);if(typeof t=="boolean"){const s=t?"true":"false";return this.d===s?qt.INSTANCE:ns.INSTANCE}return this}evaluate(t){return t.getValue(this.c)!=this.d}serialize(){return`${this.c} != '${this.d}'`}keys(){return[this.c]}map(t){return t.mapNotEquals(this.c,this.d)}negate(){return this.f||(this.f=yo.create(this.c,this.d,this)),this.f}},mn=class j6{static create(t,s=null){const i=Et.get(t);return typeof i=="boolean"?i?qt.INSTANCE:ns.INSTANCE:new j6(t,s)}constructor(t,s){this.c=t,this.d=s,this.type=3}cmp(t){return t.type!==this.type?this.type-t.type:ym(this.c,t.c)}equals(t){return t.type===this.type?this.c===t.c:!1}substituteConstants(){const t=Et.get(this.c);return typeof t=="boolean"?t?qt.INSTANCE:ns.INSTANCE:this}evaluate(t){return!t.getValue(this.c)}serialize(){return`!${this.c}`}keys(){return[this.c]}map(t){return t.mapNot(this.c)}negate(){return this.d||(this.d=pn.create(this.c,this)),this.d}};function Ja(e,t){if(typeof e=="string"){const s=parseFloat(e);isNaN(s)||(e=s)}return typeof e=="string"||typeof e=="number"?t(e):qt.INSTANCE}var Qa=class _6{static create(t,s,i=null){return Ja(s,r=>new _6(t,r,i))}constructor(t,s,i){this.c=t,this.d=s,this.f=i,this.type=12}cmp(t){return t.type!==this.type?this.type-t.type:Lr(this.c,this.d,t.c,t.d)}equals(t){return t.type===this.type?this.c===t.c&&this.d===t.d:!1}substituteConstants(){return this}evaluate(t){return typeof this.d=="string"?!1:parseFloat(t.getValue(this.c))>this.d}serialize(){return`${this.c} > ${this.d}`}keys(){return[this.c]}map(t){return t.mapGreater(this.c,this.d)}negate(){return this.f||(this.f=Xa.create(this.c,this.d,this)),this.f}},Za=class B6{static create(t,s,i=null){return Ja(s,r=>new B6(t,r,i))}constructor(t,s,i){this.c=t,this.d=s,this.f=i,this.type=13}cmp(t){return t.type!==this.type?this.type-t.type:Lr(this.c,this.d,t.c,t.d)}equals(t){return t.type===this.type?this.c===t.c&&this.d===t.d:!1}substituteConstants(){return this}evaluate(t){return typeof this.d=="string"?!1:parseFloat(t.getValue(this.c))>=this.d}serialize(){return`${this.c} >= ${this.d}`}keys(){return[this.c]}map(t){return t.mapGreaterEquals(this.c,this.d)}negate(){return this.f||(this.f=Ya.create(this.c,this.d,this)),this.f}},Ya=class z6{static create(t,s,i=null){return Ja(s,r=>new z6(t,r,i))}constructor(t,s,i){this.c=t,this.d=s,this.f=i,this.type=14}cmp(t){return t.type!==this.type?this.type-t.type:Lr(this.c,this.d,t.c,t.d)}equals(t){return t.type===this.type?this.c===t.c&&this.d===t.d:!1}substituteConstants(){return this}evaluate(t){return typeof this.d=="string"?!1:parseFloat(t.getValue(this.c))<this.d}serialize(){return`${this.c} < ${this.d}`}keys(){return[this.c]}map(t){return t.mapSmaller(this.c,this.d)}negate(){return this.f||(this.f=Za.create(this.c,this.d,this)),this.f}},Xa=class H6{static create(t,s,i=null){return Ja(s,r=>new H6(t,r,i))}constructor(t,s,i){this.c=t,this.d=s,this.f=i,this.type=15}cmp(t){return t.type!==this.type?this.type-t.type:Lr(this.c,this.d,t.c,t.d)}equals(t){return t.type===this.type?this.c===t.c&&this.d===t.d:!1}substituteConstants(){return this}evaluate(t){return typeof this.d=="string"?!1:parseFloat(t.getValue(this.c))<=this.d}serialize(){return`${this.c} <= ${this.d}`}keys(){return[this.c]}map(t){return t.mapSmallerEquals(this.c,this.d)}negate(){return this.f||(this.f=Qa.create(this.c,this.d,this)),this.f}},xo=class V6{static create(t,s){return new V6(t,s)}constructor(t,s){this.d=t,this.f=s,this.type=7,this.c=null}cmp(t){if(t.type!==this.type)return this.type-t.type;if(this.d<t.d)return-1;if(this.d>t.d)return 1;const s=this.f?this.f.source:"",i=t.f?t.f.source:"";return s<i?-1:s>i?1:0}equals(t){if(t.type===this.type){const s=this.f?this.f.source:"",i=t.f?t.f.source:"";return this.d===t.d&&s===i}return!1}substituteConstants(){return this}evaluate(t){const s=t.getValue(this.d);return this.f?this.f.test(s):!1}serialize(){const t=this.f?`/${this.f.source}/${this.f.flags}`:"/invalid/";return`${this.d} =~ ${t}`}keys(){return[this.d]}map(t){return t.mapRegex(this.d,this.f)}negate(){return this.c||(this.c=w2.create(this)),this.c}},w2=class J1{static create(t){return new J1(t)}constructor(t){this.c=t,this.type=8}cmp(t){return t.type!==this.type?this.type-t.type:this.c.cmp(t.c)}equals(t){return t.type===this.type?this.c.equals(t.c):!1}substituteConstants(){return this}evaluate(t){return!this.c.evaluate(t)}serialize(){return`!(${this.c.serialize()})`}keys(){return this.c.keys()}map(t){return new J1(this.c.map(t))}negate(){return this.c}};function wm(e){let t=null;for(let s=0,i=e.length;s<i;s++){const r=e[s].substituteConstants();if(e[s]!==r&&t===null){t=[];for(let n=0;n<s;n++)t[n]=e[n]}t!==null&&(t[s]=r)}return t===null?e:t}var bm=class ro{static create(t,s,i){return ro.d(t,s,i)}constructor(t,s){this.expr=t,this.c=s,this.type=6}cmp(t){if(t.type!==this.type)return this.type-t.type;if(this.expr.length<t.expr.length)return-1;if(this.expr.length>t.expr.length)return 1;for(let s=0,i=this.expr.length;s<i;s++){const r=bo(this.expr[s],t.expr[s]);if(r!==0)return r}return 0}equals(t){if(t.type===this.type){if(this.expr.length!==t.expr.length)return!1;for(let s=0,i=this.expr.length;s<i;s++)if(!this.expr[s].equals(t.expr[s]))return!1;return!0}return!1}substituteConstants(){const t=wm(this.expr);return t===this.expr?this:ro.create(t,this.c,!1)}evaluate(t){for(let s=0,i=this.expr.length;s<i;s++)if(!this.expr[s].evaluate(t))return!1;return!0}static d(t,s,i){const r=[];let n=!1;for(const o of t)if(o){if(o.type===1){n=!0;continue}if(o.type===0)return qt.INSTANCE;if(o.type===6){r.push(...o.expr);continue}r.push(o)}if(r.length===0&&n)return ns.INSTANCE;if(r.length!==0){if(r.length===1)return r[0];r.sort(bo);for(let o=1;o<r.length;o++)r[o-1].equals(r[o])&&(r.splice(o,1),o--);if(r.length===1)return r[0];for(;r.length>1;){const o=r[r.length-1];if(o.type!==9)break;r.pop();const a=r.pop(),c=r.length===0,l=nh.create(o.expr.map(u=>ro.create([u,a],null,i)),null,c);l&&(r.push(l),r.sort(bo))}if(r.length===1)return r[0];if(i){for(let o=0;o<r.length;o++)for(let a=o+1;a<r.length;a++)if(r[o].negate().equals(r[a]))return qt.INSTANCE;if(r.length===1)return r[0]}return new ro(r,s)}}serialize(){return this.expr.map(t=>t.serialize()).join(" && ")}keys(){const t=[];for(const s of this.expr)t.push(...s.keys());return t}map(t){return new ro(this.expr.map(s=>s.map(t)),null)}negate(){if(!this.c){const t=[];for(const s of this.expr)t.push(s.negate());this.c=nh.create(t,this,!0)}return this.c}},nh=class Zr{static create(t,s,i){return Zr.d(t,s,i)}constructor(t,s){this.expr=t,this.c=s,this.type=9}cmp(t){if(t.type!==this.type)return this.type-t.type;if(this.expr.length<t.expr.length)return-1;if(this.expr.length>t.expr.length)return 1;for(let s=0,i=this.expr.length;s<i;s++){const r=bo(this.expr[s],t.expr[s]);if(r!==0)return r}return 0}equals(t){if(t.type===this.type){if(this.expr.length!==t.expr.length)return!1;for(let s=0,i=this.expr.length;s<i;s++)if(!this.expr[s].equals(t.expr[s]))return!1;return!0}return!1}substituteConstants(){const t=wm(this.expr);return t===this.expr?this:Zr.create(t,this.c,!1)}evaluate(t){for(let s=0,i=this.expr.length;s<i;s++)if(this.expr[s].evaluate(t))return!0;return!1}static d(t,s,i){let r=[],n=!1;if(t){for(let o=0,a=t.length;o<a;o++){const c=t[o];if(c){if(c.type===0){n=!0;continue}if(c.type===1)return ns.INSTANCE;if(c.type===9){r=r.concat(c.expr);continue}r.push(c)}}if(r.length===0&&n)return qt.INSTANCE;r.sort(bo)}if(r.length!==0){if(r.length===1)return r[0];for(let o=1;o<r.length;o++)r[o-1].equals(r[o])&&(r.splice(o,1),o--);if(r.length===1)return r[0];if(i){for(let o=0;o<r.length;o++)for(let a=o+1;a<r.length;a++)if(r[o].negate().equals(r[a]))return ns.INSTANCE;if(r.length===1)return r[0]}return new Zr(r,s)}}serialize(){return this.expr.map(t=>t.serialize()).join(" || ")}keys(){const t=[];for(const s of this.expr)t.push(...s.keys());return t}map(t){return new Zr(this.expr.map(s=>s.map(t)),null)}negate(){if(!this.c){const t=[];for(const s of this.expr)t.push(s.negate());for(;t.length>1;){const s=t.shift(),i=t.shift(),r=[];for(const n of Em(s))for(const o of Em(i))r.push(bm.create([n,o],null,!1));t.unshift(Zr.create(r,null,!1))}this.c=Zr.create(t,this,!0)}return this.c}},Te=class vu extends pn{static{this.d=[]}static all(){return vu.d.values()}constructor(t,s,i){super(t,null),this.f=s,typeof i=="object"?vu.d.push({...i,key:t}):i!==!0&&vu.d.push({key:t,description:i,type:s!=null?typeof s:void 0})}bindTo(t){return t.createKey(this.key,this.f)}getValue(t){return t.getContextKeyValue(this.key)}toNegated(){return this.negate()}isEqualTo(t){return yo.create(this.key,t)}notEqualsTo(t){return Eo.create(this.key,t)}},b2=he("contextKeyService");function ym(e,t){return e<t?-1:e>t?1:0}function Lr(e,t,s,i){return e<s?-1:e>s?1:t<i?-1:t>i?1:0}function Em(e){return e.type===9?e.expr:[e]}var de=he("logService"),Ka=he("loggerService");function oh(e){return Xi(e)}var Q;(function(e){e[e.Off=0]="Off",e[e.Trace=1]="Trace",e[e.Debug=2]="Debug",e[e.Info=3]="Info",e[e.Warning=4]="Warning",e[e.Error=5]="Error"})(Q||(Q={}));var ec=Q.Info;function y2(e,t,s){switch(t){case Q.Trace:e.trace(s);break;case Q.Debug:e.debug(s);break;case Q.Info:e.info(s);break;case Q.Warning:e.warn(s);break;case Q.Error:e.error(s);break;case Q.Off:break;default:throw new Error(`Invalid log level ${t}`)}}function gn(e,t=!1){let s="";for(let i=0;i<e.length;i++){let r=e[i];if(r instanceof Error&&(r=er(r,t)),typeof r=="object")try{r=JSON.stringify(r)}catch{}s+=(i>0?" ":"")+r}return s}var tc=class extends se{constructor(){super(...arguments),this.b=ec,this.c=this.B(new M),this.onDidChangeLogLevel=this.c.event}setLevel(e){this.b!==e&&(this.b=e,this.c.fire(this.b))}getLevel(){return this.b}f(e){return this.b!==Q.Off&&this.b<=e}},E2=class extends tc{constructor(e){super(),this.h=e}f(e){return this.h||super.f(e)}trace(e,...t){this.f(Q.Trace)&&this.g(Q.Trace,gn([e,...t],!0))}debug(e,...t){this.f(Q.Debug)&&this.g(Q.Debug,gn([e,...t]))}info(e,...t){this.f(Q.Info)&&this.g(Q.Info,gn([e,...t]))}warn(e,...t){this.f(Q.Warning)&&this.g(Q.Warning,gn([e,...t]))}error(e,...t){if(this.f(Q.Error))if(e instanceof Error){const s=Array.prototype.slice.call(arguments);s[0]=e.stack,this.g(Q.Error,gn(s))}else this.g(Q.Error,gn([e,...t]))}flush(){}},x2=class extends tc{constructor(e=ec,t=!0){super(),this.g=t,this.setLevel(e)}trace(e,...t){this.f(Q.Trace)&&(this.g?console.log("%cTRACE","color: #888",e,...t):console.log(e,...t))}debug(e,...t){this.f(Q.Debug)&&(this.g?console.log("%cDEBUG","background: #eee; color: #888",e,...t):console.log(e,...t))}info(e,...t){this.f(Q.Info)&&(this.g?console.log("%c INFO","color: #33f",e,...t):console.log(e,...t))}warn(e,...t){this.f(Q.Warning)&&(this.g?console.log("%c WARN","color: #993",e,...t):console.log(e,...t))}error(e,...t){this.f(Q.Error)&&(this.g?console.log("%c  ERR","color: #f33",e,...t):console.error(e,...t))}flush(){}},k2=class extends tc{constructor(e){super(),this.g=e,e.length&&this.setLevel(e[0].getLevel())}setLevel(e){for(const t of this.g)t.setLevel(e);super.setLevel(e)}trace(e,...t){for(const s of this.g)s.trace(e,...t)}debug(e,...t){for(const s of this.g)s.debug(e,...t)}info(e,...t){for(const s of this.g)s.info(e,...t)}warn(e,...t){for(const s of this.g)s.warn(e,...t)}error(e,...t){for(const s of this.g)s.error(e,...t)}flush(){for(const e of this.g)e.flush()}dispose(){for(const e of this.g)e.dispose();super.dispose()}},$2=class extends se{constructor(e,t,s){if(super(),this.h=e,this.j=t,this.b=new yt,this.c=this.B(new M),this.onDidChangeLoggers=this.c.event,this.f=this.B(new M),this.onDidChangeLogLevel=this.f.event,this.g=this.B(new M),this.onDidChangeVisibility=this.g.event,s)for(const i of s)this.b.set(i.resource,{logger:void 0,info:i})}m(e){return we(e)?[...this.b.values()].find(t=>t.info.id===e):this.b.get(e)}getLogger(e){return this.m(e)?.logger}createLogger(e,t){const s=this.n(e),i=we(e)?e:t?.id??fn(s.toString()).toString(16);let r=this.b.get(s)?.logger;const n=t?.logLevel==="always"?Q.Trace:t?.logLevel;r||(r=this.s(s,n??this.getLogLevel(s)??this.h,{...t,id:i}));const o={logger:r,info:{resource:s,id:i,logLevel:n,name:t?.name,hidden:t?.hidden,extensionId:t?.extensionId,when:t?.when}};return this.registerLogger(o.info),this.b.set(s,o),r}n(e){return we(e)?fe(this.j,`${e}.log`):e}setLogLevel(e,t){if(k.isUri(e)){const s=e,i=t,r=this.b.get(s);r&&i!==r.info.logLevel&&(r.info.logLevel=i===this.h?void 0:i,r.logger?.setLevel(i),this.b.set(r.info.resource,r),this.f.fire([s,i]))}else{this.h=e;for(const[s,i]of this.b.entries())this.b.get(s)?.info.logLevel===void 0&&i.logger?.setLevel(this.h);this.f.fire(this.h)}}setVisibility(e,t){const s=this.m(e);s&&t!==!s.info.hidden&&(s.info.hidden=!t,this.b.set(s.info.resource,s),this.g.fire([s.info.resource,t]))}getLogLevel(e){let t;return e&&(t=this.b.get(e)?.info.logLevel),t??this.h}registerLogger(e){const t=this.b.get(e.resource);t?t.info.hidden!==e.hidden&&this.setVisibility(e.resource,!e.hidden):(this.b.set(e.resource,{info:e,logger:void 0}),this.c.fire({added:[e],removed:[]}))}deregisterLogger(e){const t=this.b.get(e);t&&(t.logger&&t.logger.dispose(),this.b.delete(e),this.c.fire({added:[],removed:[t.info]}))}*getRegisteredLoggers(){for(const e of this.b.values())yield e.info}getRegisteredLogger(e){return this.b.get(e)?.info}dispose(){this.b.forEach(e=>e.logger?.dispose()),this.b.clear(),super.dispose()}},P2=class{constructor(){this.onDidChangeLogLevel=new M().event}setLevel(e){}getLevel(){return Q.Info}trace(e,...t){}debug(e,...t){}info(e,...t){}warn(e,...t){}error(e,...t){}critical(e,...t){}dispose(){}flush(){}},S2=class extends P2{};function ah(e){if(e.verbose)return Q.Trace;if(typeof e.logLevel=="string"){const t=I2(e.logLevel.toLowerCase());if(t!==void 0)return t}return ec}function xm(e){switch(e){case Q.Trace:return"trace";case Q.Debug:return"debug";case Q.Info:return"info";case Q.Warning:return"warn";case Q.Error:return"error";case Q.Off:return"off"}}function I2(e){switch(e){case"trace":return Q.Trace;case"debug":return Q.Debug;case"info":return Q.Info;case"warn":return Q.Warning;case"error":return Q.Error;case"critical":return Q.Error;case"off":return Q.Off}}var oN=new Te("logLevel",xm(Q.Info)),ot=class{constructor(e,t=[],s=!1){this.ctor=e,this.staticArguments=t,this.supportsDelayedInstantiation=s}},A2=Symbol("MicrotaskDelay");function ch(e){return!!e&&typeof e.then=="function"}function Ei(e){const t=new gi,s=e(t.token),i=new Promise((r,n)=>{const o=t.token.onCancellationRequested(()=>{o.dispose(),n(new bt)});Promise.resolve(s).then(a=>{o.dispose(),t.dispose(),r(a)},a=>{o.dispose(),t.dispose(),n(a)})});return new class{cancel(){t.cancel(),t.dispose()}then(r,n){return i.then(r,n)}catch(r){return this.then(void 0,r)}finally(r){return i.finally(r)}}}function D2(){let e,t;return{promise:new Promise((i,r)=>{e=i,t=r}),resolve:e,reject:t}}var N2=class{constructor(){this.f=!1,this.a=null,this.b=null,this.d=null}queue(e){if(this.f)return Promise.reject(new Error("Throttler is disposed"));if(this.a){if(this.d=e,!this.b){const t=()=>{if(this.b=null,this.f)return;const s=this.queue(this.d);return this.d=null,s};this.b=new Promise(s=>{this.a.then(t,t).then(s)})}return new Promise((t,s)=>{this.b.then(t,s)})}return this.a=e(),new Promise((t,s)=>{this.a.then(i=>{this.a=null,t(i)},i=>{this.a=null,s(i)})})}dispose(){this.f=!0}},R2=class{constructor(){this.a=Promise.resolve(null)}queue(e){return this.a=this.a.then(()=>e(),()=>e())}},L2=(e,t)=>{let s=!0;const i=setTimeout(()=>{s=!1,t()},e);return{isTriggered:()=>s,dispose:()=>{clearTimeout(i),s=!1}}},O2=e=>{let t=!0;return queueMicrotask(()=>{t&&(t=!1,e())}),{isTriggered:()=>t,dispose:()=>{t=!1}}},km=class{constructor(e){this.defaultDelay=e,this.a=null,this.b=null,this.d=null,this.f=null,this.g=null}trigger(e,t=this.defaultDelay){this.g=e,this.h(),this.b||(this.b=new Promise((i,r)=>{this.d=i,this.f=r}).then(()=>{if(this.b=null,this.d=null,this.g){const i=this.g;return this.g=null,i()}}));const s=()=>{this.a=null,this.d?.(null)};return this.a=t===A2?O2(s):L2(t,s),this.b}isTriggered(){return!!this.a?.isTriggered()}cancel(){this.h(),this.b&&(this.f?.(new bt),this.b=null)}h(){this.a?.dispose(),this.a=null}dispose(){this.cancel()}},vn=class{constructor(e){this.a=new km(e),this.b=new N2}trigger(e,t){return this.a.trigger(()=>this.b.queue(e),t)}isTriggered(){return this.a.isTriggered()}cancel(){this.a.cancel()}dispose(){this.a.dispose(),this.b.dispose()}},$m=class{constructor(){this.a=!1,this.b=new Promise((e,t)=>{this.d=e})}isOpen(){return this.a}open(){this.a=!0,this.d(!0)}wait(){return this.b}};function Or(e,t){return t?new Promise((s,i)=>{const r=setTimeout(()=>{n.dispose(),s()},e),n=t.onCancellationRequested(()=>{clearTimeout(r),n.dispose(),i(new bt)})}):Ei(s=>Or(e,s))}var Pm=class{constructor(e){this.a=0,this.b=!1,this.f=e,this.g=[],this.d=0,this.h=new M}whenIdle(){return this.size>0?X.toPromise(this.onDrained):Promise.resolve()}get onDrained(){return this.h.event}get size(){return this.a}queue(e){if(this.b)throw new Error("Object has been disposed");return this.a++,new Promise((t,s)=>{this.g.push({factory:e,c:t,e:s}),this.j()})}j(){for(;this.g.length&&this.d<this.f;){const e=this.g.shift();this.d++;const t=e.factory();t.then(e.c,e.e),t.then(()=>this.k(),()=>this.k())}}k(){this.b||(this.d--,--this.a===0&&this.h.fire(),this.g.length>0&&this.j())}clear(){if(this.b)throw new Error("Object has been disposed");this.g.length=0,this.a=this.d}dispose(){this.b=!0,this.g.length=0,this.a=0,this.h.dispose()}},ko=class extends Pm{constructor(){super(1)}},Sm=class{constructor(){this.a=new Map,this.b=new Set,this.d=void 0,this.f=0}async whenDrained(){if(this.g())return;const e=new lh;return this.b.add(e),e.p}g(){for(const[,e]of this.a)if(e.size>0)return!1;return!0}queueSize(e,t=ke){const s=t.getComparisonKey(e);return this.a.get(s)?.size??0}queueFor(e,t,s=ke){const i=s.getComparisonKey(e);let r=this.a.get(i);if(!r){r=new ko;const n=this.f++,o=X.once(r.onDrained)(()=>{r?.dispose(),this.a.delete(i),this.h(),this.d?.deleteAndDispose(n),this.d?.size===0&&(this.d.dispose(),this.d=void 0)});this.d||(this.d=new Tu),this.d.set(n,o),this.a.set(i,r)}return r.queue(t)}h(){this.g()&&this.j()}j(){for(const e of this.b)e.complete();this.b.clear()}dispose(){for(const[,e]of this.a)e.dispose();this.a.clear(),this.j(),this.d?.dispose()}},sc=class{constructor(e,t){this.b=-1,this.a=e,this.d=t,this.f=this.g.bind(this)}dispose(){this.cancel(),this.a=null}cancel(){this.isScheduled()&&(clearTimeout(this.b),this.b=-1)}schedule(e=this.d){this.cancel(),this.b=setTimeout(this.f,e)}get delay(){return this.d}set delay(e){this.d=e}isScheduled(){return this.b!==-1}flush(){this.isScheduled()&&(this.cancel(),this.h())}g(){this.b=-1,this.a&&this.h()}h(){this.a?.()}},Im=class{constructor(e,t){t%1e3!==0&&console.warn(`ProcessTimeRunOnceScheduler resolution is 1s, ${t}ms is not a multiple of 1000ms.`),this.a=e,this.b=t,this.d=0,this.f=-1,this.g=this.h.bind(this)}dispose(){this.cancel(),this.a=null}cancel(){this.isScheduled()&&(clearInterval(this.f),this.f=-1)}schedule(e=this.b){e%1e3!==0&&console.warn(`ProcessTimeRunOnceScheduler resolution is 1s, ${e}ms is not a multiple of 1000ms.`),this.cancel(),this.d=Math.ceil(e/1e3),this.f=setInterval(this.g,1e3)}isScheduled(){return this.f!==-1}h(){this.d--,!(this.d>0)&&(clearInterval(this.f),this.f=-1,this.a?.())}},C2=class extends sc{constructor(e,t){super(e,t),this.j=[]}work(e){this.j.push(e),this.isScheduled()||this.schedule()}h(){const e=this.j;this.j=[],this.a?.(e)}dispose(){this.j=[],super.dispose()}},M2=class extends se{constructor(e,t){super(),this.g=e,this.h=t,this.a=[],this.b=this.B(new Fu),this.f=!1}get pending(){return this.a.length}work(e){if(this.f)return!1;if(typeof this.g.maxBufferedWork=="number"){if(this.b.value){if(this.pending+e.length>this.g.maxBufferedWork)return!1}else if(this.pending+e.length-this.g.maxWorkChunkSize>this.g.maxBufferedWork)return!1}for(const t of e)this.a.push(t);return this.b.value||this.j(),!0}j(){this.h(this.a.splice(0,this.g.maxWorkChunkSize)),this.a.length>0&&(this.b.value=new sc(()=>{this.b.clear(),this.j()},this.g.throttleDelay),this.b.value.schedule())}dispose(){super.dispose(),this.f=!0}},Am,ic;(function(){typeof globalThis.requestIdleCallback!="function"||typeof globalThis.cancelIdleCallback!="function"?ic=(e,t)=>{I9(()=>{if(s)return;const i=Date.now()+15;t(Object.freeze({didTimeout:!0,timeRemaining(){return Math.max(0,i-Date.now())}}))});let s=!1;return{dispose(){s||(s=!0)}}}:ic=(e,t,s)=>{const i=e.requestIdleCallback(t,typeof s=="number"?{timeout:s}:void 0);let r=!1;return{dispose(){r||(r=!0,e.cancelIdleCallback(i))}}},Am=e=>ic(globalThis,e)})();var F2=class{constructor(e,t){this.g=!1,this.d=()=>{try{this.j=t()}catch(s){this.l=s}finally{this.g=!0}},this.f=ic(e,()=>this.d())}dispose(){this.f.dispose()}get value(){if(this.g||(this.f.dispose(),this.d()),this.l)throw this.l;return this.j}get isInitialized(){return this.g}},T2=class extends F2{constructor(e){super(globalThis,e)}};async function U2(e,t,s){let i;for(let r=0;r<s;r++)try{return await e()}catch(n){i=n,await Or(t)}throw i}var Dm;(function(e){e[e.Resolved=0]="Resolved",e[e.Rejected=1]="Rejected"})(Dm||(Dm={}));var lh=class{get isRejected(){return this.d?.outcome===1}get isResolved(){return this.d?.outcome===0}get isSettled(){return!!this.d}get value(){return this.d?.outcome===0?this.d?.value:void 0}constructor(){this.p=new Promise((e,t)=>{this.a=e,this.b=t})}complete(e){return new Promise(t=>{this.a(e),this.d={outcome:0,value:e},t()})}error(e){return new Promise(t=>{this.b(e),this.d={outcome:1,value:e},t()})}cancel(){return this.error(new bt)}},jt;(function(e){async function t(i){let r;const n=await Promise.all(i.map(o=>o.then(a=>a,a=>{r||(r=a)})));if(typeof r<"u")throw r;return n}e.settled=t;function s(i){return new Promise(async(r,n)=>{try{await i(r,n)}catch(o){n(o)}})}e.withAsyncBody=s})(jt||(jt={}));var Nm;(function(e){e[e.Initial=0]="Initial",e[e.DoneOK=1]="DoneOK",e[e.DoneError=2]="DoneError"})(Nm||(Nm={}));var aN=class vs{static fromArray(t){return new vs(s=>{s.emitMany(t)})}static fromPromise(t){return new vs(async s=>{s.emitMany(await t)})}static fromPromises(t){return new vs(async s=>{await Promise.all(t.map(async i=>s.emitOne(await i)))})}static merge(t){return new vs(async s=>{await Promise.all(t.map(async i=>{for await(const r of i)s.emitOne(r)}))})}static{this.EMPTY=vs.fromArray([])}constructor(t,s){this.a=0,this.b=[],this.d=null,this.f=s,this.g=new M,queueMicrotask(async()=>{const i={emitOne:r=>this.h(r),emitMany:r=>this.j(r),reject:r=>this.l(r)};try{await Promise.resolve(t(i)),this.k()}catch(r){this.l(r)}finally{i.emitOne=void 0,i.emitMany=void 0,i.reject=void 0}})}[Symbol.asyncIterator](){let t=0;return{next:async()=>{do{if(this.a===2)throw this.d;if(t<this.b.length)return{done:!1,value:this.b[t++]};if(this.a===1)return{done:!0,value:void 0};await X.toPromise(this.g.event)}while(!0)},return:async()=>(this.f?.(),{done:!0,value:void 0})}}static map(t,s){return new vs(async i=>{for await(const r of t)i.emitOne(s(r))})}map(t){return vs.map(this,t)}static filter(t,s){return new vs(async i=>{for await(const r of t)s(r)&&i.emitOne(r)})}filter(t){return vs.filter(this,t)}static coalesce(t){return vs.filter(t,s=>!!s)}coalesce(){return vs.coalesce(this)}static async toPromise(t){const s=[];for await(const i of t)s.push(i);return s}toPromise(){return vs.toPromise(this)}h(t){this.a===0&&(this.b.push(t),this.g.fire())}j(t){this.a===0&&(this.b=this.b.concat(t),this.g.fire())}k(){this.a===0&&(this.a=1,this.g.fire())}l(t){this.a===0&&(this.a=2,this.d=t,this.g.fire())}};function rc(e){const t=e;return t?[t.on,t.pause,t.resume,t.destroy].every(s=>typeof s=="function"):!1}function uh(e){const t=e;return t?rc(t.stream)&&Array.isArray(t.buffer)&&typeof t.ended=="boolean":!1}function nc(e,t){return new j2(e,t)}var j2=class{constructor(e,t){this.e=e,this.f=t,this.a={flowing:!1,ended:!1,destroyed:!1},this.b={data:[],error:[]},this.c={data:[],error:[],end:[]},this.d=[]}pause(){this.a.destroyed||(this.a.flowing=!1)}resume(){this.a.destroyed||this.a.flowing||(this.a.flowing=!0,this.j(),this.k(),this.l())}write(e){if(!this.a.destroyed){if(this.a.flowing)this.g(e);else if(this.b.data.push(e),typeof this.f?.highWaterMark=="number"&&this.b.data.length>this.f.highWaterMark)return new Promise(t=>this.d.push(t))}}error(e){this.a.destroyed||(this.a.flowing?this.h(e):this.b.error.push(e))}end(e){this.a.destroyed||(typeof e<"u"&&this.write(e),this.a.flowing?(this.i(),this.destroy()):this.a.ended=!0)}g(e){this.c.data.slice(0).forEach(t=>t(e))}h(e){this.c.error.length===0?Ut(e):this.c.error.slice(0).forEach(t=>t(e))}i(){this.c.end.slice(0).forEach(e=>e())}on(e,t){if(!this.a.destroyed)switch(e){case"data":this.c.data.push(t),this.resume();break;case"end":this.c.end.push(t),this.a.flowing&&this.l()&&this.destroy();break;case"error":this.c.error.push(t),this.a.flowing&&this.k();break}}removeListener(e,t){if(this.a.destroyed)return;let s;switch(e){case"data":s=this.c.data;break;case"end":s=this.c.end;break;case"error":s=this.c.error;break}if(s){const i=s.indexOf(t);i>=0&&s.splice(i,1)}}j(){if(this.b.data.length>0){const e=this.e(this.b.data);this.g(e),this.b.data.length=0;const t=[...this.d];this.d.length=0,t.forEach(s=>s())}}k(){if(this.c.error.length>0){for(const e of this.b.error)this.h(e);this.b.error.length=0}}l(){return this.a.ended?(this.i(),this.c.end.length>0):!1}destroy(){this.a.destroyed||(this.a.destroyed=!0,this.a.ended=!0,this.b.data.length=0,this.b.error.length=0,this.c.data.length=0,this.c.error.length=0,this.c.end.length=0,this.d.length=0)}};function _2(e,t){const s=[];let i;for(;(i=e.read())!==null;)s.push(i);return t(s)}function B2(e,t,s){const i=[];let r;for(;(r=e.read())!==null&&i.length<s;)i.push(r);return r===null&&i.length>0?t(i):{read:()=>{if(i.length>0)return i.shift();if(typeof r<"u"){const n=r;return r=void 0,n}return e.read()}}}function Rm(e,t){return new Promise((s,i)=>{const r=[];oc(e,{onData:n=>{t&&r.push(n)},onError:n=>{t?i(n):s(void 0)},onEnd:()=>{s(t?t(r):void 0)}})})}function oc(e,t,s){e.on("error",i=>{s?.isCancellationRequested||t.onError(i)}),e.on("end",()=>{s?.isCancellationRequested||t.onEnd()}),e.on("data",i=>{s?.isCancellationRequested||t.onData(i)})}function z2(e,t){return new Promise((s,i)=>{const r=new at,n=[],o=l=>{if(n.push(l),n.length>t)return r.dispose(),e.pause(),s({stream:e,buffer:n,ended:!1})},a=l=>(r.dispose(),i(l)),c=()=>(r.dispose(),s({stream:e,buffer:n,ended:!0}));r.add(ge(()=>e.removeListener("error",a))),e.on("error",a),r.add(ge(()=>e.removeListener("end",c))),e.on("end",c),r.add(ge(()=>e.removeListener("data",o))),e.on("data",o)})}function H2(e){let t=!1;return{read:()=>t?null:(t=!0,e)}}function Lm(e,t,s){const i=nc(s);return oc(e,{onData:r=>i.write(t.data(r)),onError:r=>i.error(t.error?t.error(r):r),onEnd:()=>i.end()}),i}var ac=typeof Buffer<"u",V2=new ln(()=>new Uint8Array(256)),hh,fh,j=class Is{static alloc(t){return ac?new Is(Buffer.allocUnsafe(t)):new Is(new Uint8Array(t))}static wrap(t){return ac&&!Buffer.isBuffer(t)&&(t=Buffer.from(t.buffer,t.byteOffset,t.byteLength)),new Is(t)}static fromString(t,s){return!(s?.dontUseNodeBuffer||!1)&&ac?new Is(Buffer.from(t)):(hh||(hh=new TextEncoder),new Is(hh.encode(t)))}static fromByteArray(t){const s=Is.alloc(t.length);for(let i=0,r=t.length;i<r;i++)s.buffer[i]=t[i];return s}static concat(t,s){if(typeof s>"u"){s=0;for(let n=0,o=t.length;n<o;n++)s+=t[n].byteLength}const i=Is.alloc(s);let r=0;for(let n=0,o=t.length;n<o;n++){const a=t[n];i.set(a,r),r+=a.byteLength}return i}constructor(t){this.buffer=t,this.byteLength=this.buffer.byteLength}clone(){const t=Is.alloc(this.byteLength);return t.set(this),t}toString(){return ac?this.buffer.toString():(fh||(fh=new TextDecoder),fh.decode(this.buffer))}slice(t,s){return new Is(this.buffer.subarray(t,s))}set(t,s){if(t instanceof Is)this.buffer.set(t.buffer,s);else if(t instanceof Uint8Array)this.buffer.set(t,s);else if(t instanceof ArrayBuffer)this.buffer.set(new Uint8Array(t),s);else if(ArrayBuffer.isView(t))this.buffer.set(new Uint8Array(t.buffer,t.byteOffset,t.byteLength),s);else throw new Error("Unknown argument 'array'")}readUInt32BE(t){return q2(this.buffer,t)}writeUInt32BE(t,s){G2(this.buffer,t,s)}readUInt32LE(t){return J2(this.buffer,t)}writeUInt32LE(t,s){Q2(this.buffer,t,s)}readUInt8(t){return Z2(this.buffer,t)}writeUInt8(t,s){Y2(this.buffer,t,s)}indexOf(t,s=0){return W2(this.buffer,t instanceof Is?t.buffer:t,s)}};function W2(e,t,s=0){const i=t.byteLength,r=e.byteLength;if(i===0)return 0;if(i===1)return e.indexOf(t[0]);if(i>r-s)return-1;const n=V2.value;n.fill(t.length);for(let l=0;l<t.length;l++)n[t[l]]=t.length-l-1;let o=s+t.length-1,a=o,c=-1;for(;o<r;)if(e[o]===t[a]){if(a===0){c=o;break}o--,a--}else o+=Math.max(t.length-a,n[e[o]]),a=t.length-1;return c}function q2(e,t){return e[t]*2**24+e[t+1]*2**16+e[t+2]*2**8+e[t+3]}function G2(e,t,s){e[s+3]=t,t=t>>>8,e[s+2]=t,t=t>>>8,e[s+1]=t,t=t>>>8,e[s]=t}function J2(e,t){return e[t+0]<<0>>>0|e[t+1]<<8>>>0|e[t+2]<<16>>>0|e[t+3]<<24>>>0}function Q2(e,t,s){e[s+0]=t&255,t=t>>>8,e[s+1]=t&255,t=t>>>8,e[s+2]=t&255,t=t>>>8,e[s+3]=t&255}function Z2(e,t){return e[t]}function Y2(e,t,s){e[s]=t}function X2(e){return _2(e,t=>j.concat(t))}function K2(e){return H2(e)}function xi(e){return Rm(e,t=>j.concat(t))}async function eE(e){return e.ended?j.concat(e.buffer):j.concat([...e.buffer,await xi(e.stream)])}function tE(e){return Lm(e,{data:t=>typeof t=="string"?j.fromString(t):j.wrap(t)},t=>j.concat(t))}function sE(e){return nc(t=>j.concat(t),e)}function Om(e){let t=0,s=0,i=0;const r=new Uint8Array(Math.floor(e.length/4*3)),n=a=>{switch(s){case 3:r[i++]=t|a,s=0;break;case 2:r[i++]=t|a>>>2,t=a<<6,s=3;break;case 1:r[i++]=t|a>>>4,t=a<<4,s=2;break;default:t=a<<2,s=1}};for(let a=0;a<e.length;a++){const c=e.charCodeAt(a);if(c>=65&&c<=90)n(c-65);else if(c>=97&&c<=122)n(c-97+26);else if(c>=48&&c<=57)n(c-48+52);else if(c===43||c===45)n(62);else if(c===47||c===95)n(63);else{if(c===61)break;throw new SyntaxError(`Unexpected base64 character ${e[a]}`)}}const o=i;for(;s>0;)n(0);return j.wrap(r).slice(0,o)}var iE="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/",rE="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";function dh({buffer:e},t=!0,s=!1){const i=s?rE:iE;let r="";const n=e.byteLength%3;let o=0;for(;o<e.byteLength-n;o+=3){const a=e[o+0],c=e[o+1],l=e[o+2];r+=i[a>>>2],r+=i[(a<<4|c>>>4)&63],r+=i[(c<<2|l>>>6)&63],r+=i[l&63]}if(n===1){const a=e[o+0];r+=i[a>>>2],r+=i[a<<4&63],t&&(r+="==")}else if(n===2){const a=e[o+0],c=e[o+1];r+=i[a>>>2],r+=i[(a<<4|c>>>4)&63],r+=i[c<<2&63],t&&(r+="=")}return r}var Cm;(function(e){e[e.None=0]="None",e[e.UnexpectedEndOfComment=1]="UnexpectedEndOfComment",e[e.UnexpectedEndOfString=2]="UnexpectedEndOfString",e[e.UnexpectedEndOfNumber=3]="UnexpectedEndOfNumber",e[e.InvalidUnicode=4]="InvalidUnicode",e[e.InvalidEscapeCharacter=5]="InvalidEscapeCharacter",e[e.InvalidCharacter=6]="InvalidCharacter"})(Cm||(Cm={}));var Mm;(function(e){e[e.OpenBraceToken=1]="OpenBraceToken",e[e.CloseBraceToken=2]="CloseBraceToken",e[e.OpenBracketToken=3]="OpenBracketToken",e[e.CloseBracketToken=4]="CloseBracketToken",e[e.CommaToken=5]="CommaToken",e[e.ColonToken=6]="ColonToken",e[e.NullKeyword=7]="NullKeyword",e[e.TrueKeyword=8]="TrueKeyword",e[e.FalseKeyword=9]="FalseKeyword",e[e.StringLiteral=10]="StringLiteral",e[e.NumericLiteral=11]="NumericLiteral",e[e.LineCommentTrivia=12]="LineCommentTrivia",e[e.BlockCommentTrivia=13]="BlockCommentTrivia",e[e.LineBreakTrivia=14]="LineBreakTrivia",e[e.Trivia=15]="Trivia",e[e.Unknown=16]="Unknown",e[e.EOF=17]="EOF"})(Mm||(Mm={}));var Fm;(function(e){e[e.InvalidSymbol=1]="InvalidSymbol",e[e.InvalidNumberFormat=2]="InvalidNumberFormat",e[e.PropertyNameExpected=3]="PropertyNameExpected",e[e.ValueExpected=4]="ValueExpected",e[e.ColonExpected=5]="ColonExpected",e[e.CommaExpected=6]="CommaExpected",e[e.CloseBraceExpected=7]="CloseBraceExpected",e[e.CloseBracketExpected=8]="CloseBracketExpected",e[e.EndOfFileExpected=9]="EndOfFileExpected",e[e.InvalidCommentToken=10]="InvalidCommentToken",e[e.UnexpectedEndOfComment=11]="UnexpectedEndOfComment",e[e.UnexpectedEndOfString=12]="UnexpectedEndOfString",e[e.UnexpectedEndOfNumber=13]="UnexpectedEndOfNumber",e[e.InvalidUnicode=14]="InvalidUnicode",e[e.InvalidEscapeCharacter=15]="InvalidEscapeCharacter",e[e.InvalidCharacter=16]="InvalidCharacter"})(Fm||(Fm={}));var $o;(function(e){e.DEFAULT={allowTrailingComma:!0}})($o||($o={}));function Tm(e,t=!1){let s=0;const i=e.length;let r="",n=0,o=16,a=0;function c(y){let P=0,R=0;for(;P<y;){const N=e.charCodeAt(s);if(N>=48&&N<=57)R=R*16+N-48;else if(N>=65&&N<=70)R=R*16+N-65+10;else if(N>=97&&N<=102)R=R*16+N-97+10;else break;s++,P++}return P<y&&(R=-1),R}function l(y){s=y,r="",n=0,o=16,a=0}function u(){const y=s;if(e.charCodeAt(s)===48)s++;else for(s++;s<e.length&&wn(e.charCodeAt(s));)s++;if(s<e.length&&e.charCodeAt(s)===46)if(s++,s<e.length&&wn(e.charCodeAt(s)))for(s++;s<e.length&&wn(e.charCodeAt(s));)s++;else return a=3,e.substring(y,s);let P=s;if(s<e.length&&(e.charCodeAt(s)===69||e.charCodeAt(s)===101))if(s++,(s<e.length&&e.charCodeAt(s)===43||e.charCodeAt(s)===45)&&s++,s<e.length&&wn(e.charCodeAt(s))){for(s++;s<e.length&&wn(e.charCodeAt(s));)s++;P=s}else a=3;return e.substring(y,P)}function h(){let y="",P=s;for(;;){if(s>=i){y+=e.substring(P,s),a=2;break}const R=e.charCodeAt(s);if(R===34){y+=e.substring(P,s),s++;break}if(R===92){if(y+=e.substring(P,s),s++,s>=i){a=2;break}switch(e.charCodeAt(s++)){case 34:y+='"';break;case 92:y+="\\";break;case 47:y+="/";break;case 98:y+="\b";break;case 102:y+="\f";break;case 110:y+=`
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   

Canceled: Canceled
    at km.cancel (file:///tmp/vscode-web/out/vs/server/node/server.main.js:33:20981)
    at km.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:33:21051)
    at vn.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:33:21259)
    at Pt (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:701)
    at I6.clear (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1246)
    at I6.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1175)
    at a$.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1910)
    at Pt (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:701)
    at I6.clear (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1246)
    at I6.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1175)
    at Fu.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:2240)
    at Pt (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:701)
    at I6.clear (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1246)
    at I6.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1175)
    at c$.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1910)
    at c$.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:63:65527)
    at Pt (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:701)
    at I6.clear (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1246)
    at I6.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1175)
    at Aa.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1910)
    at Pt (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:701)
    at I6.clear (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1246)
    at I6.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1175)
    at F$.dispose (file:///tmp/vscode-web/out/vs/server/node/server.main.js:25:1910)
    at T$ (file:///tmp/vscode-web/out/vs/server/node/server.main.js:70:15944)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v20.16.0
```
